idp-core/src/main/java/io/mosip/idp/core/dto/WalletBindingRequest.java

```diff
@@ -32,8 +33,12 @@ public class WalletBindingRequest {
 
 	@NotNull(message = ErrorConstants.INVALID_CHALLENGE_LIST)
 	@Size(min = 1, max = 5, message = ErrorConstants.INVALID_CHALLENGE_LIST)
-	private List<AuthChallenge> challengeList;
+	private List<@Valid KeyBindingAuthChallenge> challengeList;
     
     @NotEmpty(message = ErrorConstants.INVALID_PUBLIC_KEY)
     private Map<String, Object> publicKey;
+
+	@NotEmpty(message = ErrorConstants.INVALID_AUTH_FACTOR_TYPE)
+	@Size(min = 1, max = 10, message = ErrorConstants.INVALID_AUTH_FACTOR_TYPE)
+	private List<String> authFactorTypes; //auth factors to allow in validate-binding request
 }
```

Comment:
But this will give an assumption that we are binding the auth type only to provided formats. On app upgrade, if a new format is introduced we should force the user to re-bind the key.

Reply:
we will not be storing the format, only validation is performed based on the list of supported formats returned by the wrapper for the given auth-factor-type.